### PR TITLE
Teleportation now uses QueueTask

### LINKED ISSUE #3389: TeleportToCoords not working after player switches world
Client version:   1.10.2
Server OS:        Linux
Commit id:        b268db4caae1b5612ae0f3adabf164efc06de368

### Expected behavior
Players should teleport to the correct coordinates specified in TeleportToCoords after switching worlds.

### Actual behavior
The player doesn't teleport to the coordinates, and stays at the spawn point. When the player doesn't have to switch worlds, it teleports to the correct coordinates.

### Steps to reproduce the behavior
Try commands like /tp, /warp in Core and Essentials.

### PR DESCRIPTION
**Pull request by LogicParrot**
Closes #3389

### CODE REVIEW

File: src/Entities/Player.cpp [hunk 1]
```diff
@@ -1447,13 +1447,16 @@ void cPlayer::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
...
 	if (!cRoot::Get()->GetPluginManager()->CallHookEntityTeleport(*this, m_LastPosition, Vector3d(a_PosX, a_PosY, a_PosZ)))
 	{
-		SetPosition(a_PosX, a_PosY, a_PosZ);
-		FreezeInternal(GetPosition(), false);
-		m_LastGroundHeight = a_PosY;
-		m_bIsTeleporting = true;
-
-		m_World->BroadcastTeleportEntity(*this, GetClientHandle());
-		m_ClientHandle->SendPlayerMoveLook();
+		m_World->QueueTask([this, a_PosX, a_PosY, a_PosZ](cWorld & a_World)
```
Comment (madmaxoft) [inline src/Entities/Player.cpp]: No, this is not good. If the player disconnects between the enqueue and the execution of the task, the server will crash. You cannot capture `this`.

Comment (LogicParrot) [inline src/Entities/Player.cpp]: Good point. This also means `DoMoveToWorld` may be currently unsafe.
I also found out `ScheduleMoveToWorld` is buggy when called by plugins.
I am working on a proper fix all 3 methods.

Can we merge this first? warps are currently broken and they are critical for many servers. The fix is worth the minor risk of a crash. I will be fixing this fully very soon.

Comment (madmaxoft) [inline src/Entities/Player.cpp]: Just capture the Player's UniqueID and use `DoWithEntityByID`

Comment (LogicParrot) [inline src/Entities/Player.cpp]: Ok.

**Pull request closed without merging**